src/web/app/pages-instructor/instructor-session-base-page.component.ts

```diff
@@ -152,10 +152,11 @@ export abstract class InstructorSessionBasePageComponent {
    * Updates the instructor privilege in {@code SessionsTableRowModel}.
    */
   protected updateInstructorPrivilege(model: SessionsTableRowModel): void {
-    this.httpRequestService.get('/instructor/privilege', {
-      courseid: model.feedbackSession.courseId,
-      fsname: model.feedbackSession.feedbackSessionName,
-    }).subscribe((instructorPrivilege: InstructorPrivilege) => {
+    this.instructorService.loadInstructorPrivilege(
+        model.feedbackSession.courseId,
+        undefined,
```

Comment:
`undefined`?

Reply:
This method has multiple optional parameters, so I am using `undefined` to skip parameters

https://stackoverflow.com/questions/8356227/skipping-optional-function-parameters-in-javascript

Reply:
Is it a good design of API?

Reply:
The other option would be to pass the `paramMap` object, but I didn't think it was a good idea since `{ [key:string]: string }` can contain other key-value pairs. 

The other option is to create custom types for the function arguments. Would that be a better option?

Reply:
You have Typescript :P Try search `typescript optional parameters`

Reply:
Haha the method that I'm calling has 2 typescript optional parameters (i.e. method signature is `  loadInstructorPrivilege(courseId: string, sectionName?: string, feedbackSessionName?: string)
`). In this case, the only optional parameter that I am using is `feedbackSessionName`, so based on TypeScript's [documentation](https://www.typescriptlang.org/docs/handbook/functions.html#optional-and-default-parameters) I should use undefined for the second parameter 😅. Correct me if I'm wrong

Reply:
I think you can also try to define the interface on the fly:
```javascript
loadInstructorPrivilege(param: { courseId: string, sectionName?: string, feedbackSessionName?: string })
```